core/server/helpers/foreach.js

```diff
@@ -22,7 +17,7 @@ module.exports = function foreach(items, options) {
     }
 
     // Exclude items which should not be visible in the theme
-    items = filterItemsByVisibility(items, options);
+    items = ghostHelperUtils.visibility.filter(items, options.hash.visibility);
```

Comment:
Think we need additional protection in this helper like in [here](https://github.com/TryGhost/Ghost/blob/2.23.3/core/server/apps/subscribers/lib/helpers/input_email.js#L14) for cases when there are no parameters passed into it?

Reply:
You mean if `options.hash.visibility` is empty? `options.hash` will always exist in case of helpers, and `visibility` property if it doesn't exist is handled in the util. Or I missed I think what you meant 😅

Reply:
Hm.. if it always exists in case of helpers it's kind of strange we have this protective code all over the place: `options.hash = options.hash || {};` :sweat_smile:

Reply:
Wondering what is it for then?

Reply:
🤔 Hmm yeah I am not sure either why the added protection, from the `Hash Arguments` section [here](http://handlebarsjs.com/block_helpers.html), it seems they should always default to `{}`. No entirely clear from these docs, only case could be if his statement is only true for block helpers, which `foreach` is, but no for simple helpers, and that is why the added protection 🤷‍♂ 

> Handlebars provides the final hash as options.hash. This makes it easier to accept a variable number of parameters, while also accepting an optional Hash. If the template provides no hash arguments, Handlebars will automatically pass an empty object ({}), so you don't need to check for the existence of hash arguments.

Reply:
Though reading above doc gives impression its true for all helpers

Reply:
Yeap.. seems like overly protective code on our side :thinking: Noted, and maybe something to look to remove in the future :+1: